src/Stratis.Bitcoin.Features.Consensus/CoinViews/DBreezeCoinView.cs

```diff
@@ -213,15 +215,25 @@ public Task SaveChangesAsync(IEnumerable<UnspentOutputs> unspentOutputs, IEnumer
 
                         this.SetBlockHash(transaction, nextBlockHash);
 
-                        all.Sort(UnspentOutputsComparer.Instance);
-                        foreach (UnspentOutputs coin in all)
+                        var toRemove = all.Where(utxo => utxo.IsPrunable == true).ToList();
+                        toRemove.Sort(UnspentOutputsComparer.Instance);
+                        for (int i = 0; i < toRemove.Count; i++)
+                        {
+                            var coin = toRemove[i];
+                            this.logger.LogTrace("Outputs of transaction ID '{0}' are prunable and will be removed from the database.", coin.TransactionId);
+
+                            transaction.RemoveKey("Coins", coin.TransactionId.ToBytes(false));
+                        }
+
+
+                        var toInsert = all.Where(utxo => utxo.IsPrunable == false).ToList();
+                        toInsert.Sort(UnspentOutputsComparer.Instance);
+                        for (int i = 0; i < toInsert.Count; i++)
```

Comment:
foreach?

Reply:
about foreach i see 2 cons
1 - performance ( see https://www.dotnetperls.com/for-foreach )

> Summary. In micro-benchmarking, introducing extra local variables with foreach-loops can impact performance. However, if those local variables are reused several times in the loop body, they can lead to a performance improvement.
> Thus:
> The for-loop is faster than the foreach-loop if the array must only be accessed once per iteration.
> 

- impossibility to know at which iteration we are
so we can't know how many items we are at or how many items are left and we can't log that information if we need, or debug easily.
I consider foreach a syntactic sugar that help readability in some scenario, but I prefer for over foreach when the loop is more than just `foreach x do ActionX`